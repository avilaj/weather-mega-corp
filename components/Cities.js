import css from 'styled-jsx/css'
import Link from 'next/link'
import { map } from 'ramda'

const citiesStyles = css`
  .sentence {
    color: #444;
  }
  .cities {
    padding: 1rem;
  }
  ul {
    padding: 0;
  }
  li {
    list-style-type: none;
    padding: 0;
    margin: 0;
    display: inline-block;
  }

  a {
    color: purple;
    text-decoration: none;
  }
  li+li:before {
    content: ", "
  }
`
const Cities = ({ cities }) => (
  <div className='cities'>
    <div className='sentence'>
      { cities && `Look the weather at one of these cities` }
      { !cities && `No cities available.` }
    </div>
    <ul>
      {
        cities && map(city => (
          <li key={city.id}>
            <Link
              as={`/${city.id}`}
              href={`/weather?city=${city.id}`}
            >
              <a>{city.name}</a>
            </Link>
          </li>
        ), cities)
      }
    </ul>
    <style jsx>{citiesStyles}</style>
  </div>
)

export default Cities
