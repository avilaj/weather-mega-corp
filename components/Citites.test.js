/* global describe it expect */
import React from 'react'
import Cities from './Cities'
import renderer from 'react-test-renderer'

describe('<Cities />', () => {
  it('renders empty when cities are not provided', () => {
    const tree = renderer
    .create(<Cities />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders cities when they are provided', () => {
    const cities = [
      { id: 'china', name: 'China' }
    ]
    const tree = renderer
    .create(<Cities cities={cities} />)
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
