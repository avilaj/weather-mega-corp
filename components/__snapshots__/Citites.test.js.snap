// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cities /> renders cities when they are provided 1`] = `
<div
  className="jsx-3008306360 cities"
>
  <div
    className="jsx-3008306360 sentence"
  >
    Look the weather at one of these cities
  </div>
  <ul
    className="jsx-3008306360"
  >
    <li
      className="jsx-3008306360"
    >
      <a
        className="jsx-3008306360"
        href="/china"
        onClick={[Function]}
      >
        China
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Cities /> renders empty when cities are not provided 1`] = `
<div
  className="jsx-3008306360 cities"
>
  <div
    className="jsx-3008306360 sentence"
  >
    No cities available.
  </div>
  <ul
    className="jsx-3008306360"
  />
</div>
`;
